### PR TITLE
[awsappsignals] Add ASG Name, Instance Id, and host name to emf attributes for ec2

### PR DESCRIPTION
**Pull request by lisguo**
# Description of the issue
Currently for Application Signals on EC2, there are no EC2 specific attributes or dimensions being passed to the emf logs exporter. This PR adds support for AutoScalingGroupName, host name, and instance id.

# Description of changes
- New translator logic for EC2 for appsignals to differentiate between onprem (generic)
  - Made changes to the emf translator to include asg, instanceid, and hostname as attributes
  - Add autoscaling group tag to resource detection processor
- Made changes to the awsappsignals processor to replace the following attribute names: `host.id` -> `EC2.InstanceId`, `ec2.tag.aws:autoscaling:groupName` -> `EC2.AutoScalingGroupName`
  - Add logic to set `HostedIn.EC2.Environment` to be customer defined, the ASG name, or default to "EC2"

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Build and tested on an EC2 instance and received the following emf log:
```
{
  "EC2.AutoScalingGroupName": "AppSignals-ASG-Test-main",
  "EC2.instanceId": "i-0df527e14b7897b47",
  "HostedIn.EC2.Environment": "AppSignals-ASG-Test-main",
  "OTelLib": "AwsSpanMetricsProcessor",
  "Operation": "GET /outgoing-http-call",
  "RemoteOperation": "GET /",
  "RemoteService": "www.amazon.com",
  "Service": "AppSignals-ASG-Test",
  "Version": "1",
  "_aws": {
      "CloudWatchMetrics": [
          {
              "Namespace": "AppSignals",
              "Dimensions": [
                  [
                      "HostedIn.EC2.Environment",
                      "Operation",
                      "Service"
                  ],
                  [
                      "HostedIn.EC2.Environment",
                      "Service"
                  ],
                  [
                      "HostedIn.EC2.Environment",
                      "Operation",
                      "RemoteOperation",
                      "RemoteService",
                      "Service"
                  ],
                  [
                      "HostedIn.EC2.Environment",
                      "RemoteService",
                      "Service"
                  ],
                  [
                      "HostedIn.EC2.Environment",
                      "RemoteOperation",
                      "RemoteService",
                      "Service"
                  ],
                  [
                      "RemoteService"
                  ]
              ],
              "Metrics": [
                  {
                      "Name": "Fault"
                  },
                  {
                      "Name": "Error"
                  },
                  {
                      "Name": "Latency",
                      "Unit": "Milliseconds"
                  }
              ]
          }
      ],
      "Timestamp": 1708555573735
  },
  "aws.span.kind": "CLIENT",
  "host.name": "ip-172-31-19-8.ec2.internal",
  "Error": {
      "Values": [
          0
      ],
      "Counts": [
          3
      ],
      "Max": 0,
      "Min": 0,
      "Count": 3,
      "Sum": 0
  },
  "Fault": {
      "Values": [
          0.9999996694817774
      ],
      "Counts": [
          3
      ],
      "Max": 1,
      "Min": 1,
      "Count": 3,
      "Sum": 3
  },
  "Latency": {
      "Values": [
          21.261353521941878,
          38.78290693770987
      ],
      "Counts": [
          2,
          1
      ],
      "Max": 38.850934,
      "Min": 21.20459,
      "Count": 3,
      "Sum": 81.343991
  }
}
```

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

File: translator/tocwconfig/sampleConfig/appsignals_and_eks_config.yaml
Comment (majanjua-amzn) [inline translator/tocwconfig/sampleConfig/appsignals_and_eks_config.yaml:1]: Do we need an equivalent file for ec2? i.e. `appsignals_and_ec2_config.yaml`

**Review by majanjua-amzn — APPROVED** [review]
LGTM

**Review by jefchien — APPROVED** [review]

File: plugins/processors/awsappsignals/internal/resolver/ec2.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	attr "github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsappsignals/internal/attributes"
+)
+
+const AttributePlatformEC2 = "EC2"
+
+// EC2HostedInAttributes renames attributes coming from the resource detection processor
+var EC2HostedInAttributes = map[string]string{
+	attr.ResourceDetectionHostId:   attr.EC2InstanceId,
+	attr.ResourceDetectionHostName: attr.ResourceDetectionHostName,
```
[the remark this answers is not in the document]
Comment (lisguo) [reply plugins/processors/awsappsignals/internal/resolver/ec2.go:19]: Yeah this doubles as allow list which gets parsed in line 39.  will include a unit test to prove this out

[the remark this answers is not in the document]
Comment (lisguo) [reply plugins/processors/awsappsignals/internal/resolver/ec2.go:19]: Good question, I think our attributeresolver.go is our generic case. I do not see host.name in the map there. I can create a follow-up to address our on-prem case

File: plugins/processors/awsappsignals/internal/resolver/ec2_test.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	resourceAttributes.PutStr(attr.ResourceDetectionHostId, "hostid")
+	resourceAttributes.PutStr(attr.ResourceDetectionHostName, "hostname")
+	resourceAttributes.PutStr(attr.ResourceDetectionASG, "asg")
+
+	resolver.Process(attributes, resourceAttributes)
+	expectedInstanceId, ok := attributes.Get(attr.EC2InstanceId)
+	assert.True(t, ok)
+	assert.Equal(t, "hostid", expectedInstanceId.AsString())
+
+	expectedHostName, ok := attributes.Get(attr.ResourceDetectionHostName)
```
Comment (lisguo) [inline plugins/processors/awsappsignals/internal/resolver/ec2_test.go:141]: added this unit test from feedback, if I removed the resource detection host name from the resolver map, then this fails

File: plugins/processors/awsappsignals/internal/resolver/ec2.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jefchien) [reply plugins/processors/awsappsignals/internal/resolver/ec2.go:19]: Got it. Would be useful to have in the comment.

**Review by jefchien — APPROVED** [review]

**Review by okankoAMZ — APPROVED** [review]
looks good!

**Pull request merged**